### PR TITLE
Improve url handling

### PR DESCRIPTION
**Pull request by fgei**
Reviewer: @subsymbolic 
Asana: https://app.asana.com/0/0/353085934262755

**Description**: Improve url handling so the app correctly loads IP while perform searches for numbers and display correctly encoded characters in the OmniBar

**Steps to test this PR**:
TEST 1
1. Search an IP address the app should load the IP
2. Search a simple number or partial IP address the app will perform a search based on the input

TEST 2
1. Perform a search using special characters
2. The OmniBar will show the right input query

###### Reviewer Checklist:
- [x] **Ensure the PR solves the problem**
- [x] **Review every line of code**
- [x] **Ensure the PR does no harm by testing the changes thoroughly**
- [x] **Get help if you're uncomfortable with any of the above!**
- [x] Determine if there are any quick wins that improve the implementation

###### PR DRI Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications (Database connections, Grafana stats, CPU)

### CODE REVIEW

File: app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java [hunk 1]
```diff
@@ -22,75 +22,93 @@ public void whenGetHomeThenReturnHomeUrl() {
...
     }
     @Test
-    public void whenGetQueryWithRandomTextThenReturnNull() {
+    public void whenUrlIsRandomTextThenGetQueryReturnNull() {
         final String url = "asd''asd/////.com/?q=duckduckgo.com";
         final String result = AppUrls.getQuery(url);
         assertNull(result);
     }
     @Test
+    public void whenUrlIdDDGWithQueryWithEncodedCharactersThenGetQueryReturnCorrectQuery() {
```
Comment (subsymbolic) [inline app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java]: Do you mean `whenUrlIs` rather than `whenUrlId`?

Comment (fgei) [inline app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java]: oh yes, typo!

File: app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java [hunk 2]
```diff
@@ -22,75 +22,93 @@ public void whenGetHomeThenReturnHomeUrl() {
...
+        final String[] urls = new String[] {
```
Comment (fgei) [inline app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java]: yeah I agree, I though it was easier to read in a loop, but actually isn't, I'll change that now

File: app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/util/UrlUtils.java [hunk 3]
```diff
@@ -3,15 +3,19 @@
...
 
+import java.util.regex.Pattern;
+
 /**
  * Created by fgei on 5/23/17.
  */
 
 public class UrlUtils {
 
+    private static final String WEB_URL_REGEX = "^(https?:\\/\\/)?([\\da-z\\.-]+\\.[a-z\\.]{2,6}|(([\\d]+[.]){3}[\\d]+))([\\/:?=&#]{1}[\\da-z\\.-]+)*[\\/\\?]?$";
```
Comment (fgei) [inline app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/util/UrlUtils.java]: yes, that's the same from iOS :)
I think we'll benefit from using the same validation, in the end it's the same across both platform and having just one implementation will allow us to find the same bugs and have the same fixes! I'll change and test mine too

File: app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java [hunk 4]
```diff
@@ -22,76 +22,83 @@ public void whenGetHomeThenReturnHomeUrl() {
...
         assertNull(result);
     }
     @Test
-    public void whenGetSearchUrlThenReturnCorrectDDGUrl() {
+    public void whenUrlIsDDGWithQueryWithEncodedCharactersThenGetQueryReturnCorrectQuery() {
+        assertEquals("1 + 1", AppUrls.getQuery("https://www.duckduckgo.com/?ko=-1&kl=wt-wt&q=1+%2B+1"));
+        assertEquals("1+1", AppUrls.getQuery("https://www.duckduckgo.com/?ko=-1&kl=wt-wt&q=1%2B1"));
+        assertEquals("1 @ % &  4", AppUrls.getQuery("https://www.duckduckgo.com/?ko=-1&kl=wt-wt&q=1+%40+%25+%26++4"));
+
+    }
```
Comment (subsymbolic) [inline app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java]: Tiny formatting one... move blank line down

**Comment by subsymbolic** [comment]
Looks good @fgei, one last formatting suggestion. Other than that, feel free to merge when you're ready.

**Pull request merged**